relayer/processor/path_processor.go

```diff
@@ -536,19 +565,76 @@ ChannelHandshakeLoop:
 	}
 }
 
-func (pp *PathProcessor) sendMessages(pathEnd *pathEndRuntime, messages []provider.RelayerMessage) error {
+func (pp *PathProcessor) prependMsgUpdateClient(src, dst *pathEndRuntime, messages *[]provider.RelayerMessage) error {
```

Comment:
It's uncommon to see pointers to slices or maps in Go as with `messages` here. But you are modifying the argument, so it's necessary. 

I generally return a new slice as part of the function call. Or modify the slice is a struct field. 

```go
// struct is a pointer
myStruct.Slice = append(myStruct.Slice, newElements...)
```

No changes requested; just wanted to point out alternate patterns.

Reply:
I didn't feel great about this either. Luckily it's already cleaned up in the next PR: https://github.com/cosmos/relayer/blob/63d6b4e2ac812a53e8fa5c8c21929b15f6e31bb0/relayer/processor/path_processor.go#L821